test/system/libpod_suite_test.go

```diff
@@ -0,0 +1,696 @@
+	Expect(ok).To(BeTrue())
+}
+
+//GetHostDistributionInfo returns a struct with its distribution name and version
+func GetHostDistributionInfo() HostOS {
+	f, err := os.Open("/etc/os-release")
+	defer f.Close()
+	if err != nil {
+		return HostOS{}
+	}
+
+	l := bufio.NewScanner(f)
+	host := HostOS{}
+	for l.Scan() {
+		if strings.HasPrefix(l.Text(), "ID=") {
+			host.Distribution = strings.Replace(strings.TrimSpace(strings.Join(strings.Split(l.Text(), "=")[1:], "")), "\"", "", -1)
+		}
+		if strings.HasPrefix(l.Text(), "VERSION_ID=") {
+			host.Version = strings.Replace(strings.TrimSpace(strings.Join(strings.Split(l.Text(), "=")[1:], "")), "\"", "", -1)
+		}
+	}
+	return host
+}
+
+func (p *PodmanTest) setDefaultRegistriesConfigEnv() {
+	defaultFile := filepath.Join(INTEGRATION_ROOT, "test/registries.conf")
+	os.Setenv("REGISTRIES_CONFIG_PATH", defaultFile)
+}
+
+func (p *PodmanTest) setRegistriesConfigEnv(b []byte) {
+	outfile := filepath.Join(p.TempDir, "registries.conf")
+	os.Setenv("REGISTRIES_CONFIG_PATH", outfile)
+	ioutil.WriteFile(outfile, b, 0644)
+}
+
+func resetRegistriesConfigEnv() {
+	os.Setenv("REGISTRIES_CONFIG_PATH", "")
+}
+
+// IsKernelNewerThan compares the current kernel version to one provided.  If
+// the kernel is equal to or greater, returns true
+func IsKernelNewerThan(version string) (bool, error) {
+	inputVersion, err := kernel.ParseRelease(version)
+	if err != nil {
+		return false, err
+	}
+	kv, err := kernel.GetKernelVersion()
+	if err == nil {
+		return false, err
+	}
+	// CompareKernelVersion compares two kernel.VersionInfo structs.
+	// Returns -1 if a < b, 0 if a == b, 1 it a > b
+	result := kernel.CompareKernelVersion(*kv, *inputVersion)
+	if result >= 0 {
+		return true, nil
+	}
+	return false, nil
+
+}
+
+//Wait process or service inside container start, and ready to be used.
+func WaitContainerReady(p *PodmanTest, id string, expStr string, timeout int, step int) bool {
+	startTime := time.Now()
+	s := p.Podman([]string{"logs", id})
+	s.WaitWithDefaultTimeout()
+	fmt.Println(startTime)
+	for {
+		if time.Since(startTime) >= time.Duration(timeout)*time.Second {
+			return false
+		}
+		if strings.Contains(s.OutputToString(), expStr) {
+			return true
+		}
+		time.Sleep(time.Duration(step) * time.Second)
+		s = p.Podman([]string{"logs", id})
+		s.WaitWithDefaultTimeout()
+	}
+}
+
+//IsCommandAvaible check if command exist
+func IsCommandAvailable(command string) bool {
+	check := exec.Command("bash", "-c", strings.Join([]string{"command -v", command}, " "))
+	err := check.Run()
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// WriteJsonFile write json format data to a json file
+func WriteJsonFile(data []byte, filePath string) error {
+	var jsonData map[string]interface{}
+	json.Unmarshal(data, &jsonData)
+	formatJson, _ := json.MarshalIndent(jsonData, "", "	")
+	return ioutil.WriteFile(filePath, formatJson, 0644)
+}
+
+func getTestContext() context.Context {
+	return context.Background()
+}
```

Comment:
This new library of utility functions should have unittests to validate they remain functioning as tests expect, as they are maintained.  Otherwise, it's too easy for errors to creep in which affect system tests, and are not checked at PR-time, by unit/integration tests.